orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/rollback/ExplicitRollback.groovy

```diff
@@ -124,10 +127,73 @@ class ExplicitRollback implements Rollback {
     }
 
     stages << disableServerGroupStage
-    stages << buildApplySourceServerGroupCapacityStage(parentStage, parentStage.mapTo(ResizeStrategy.Source))
+
+    // only restore the min if it was pinned, i.e. there was a resize
+    if (resizeStage != null) {
+      stages << buildApplySourceServerGroupCapacityStage(parentStage, parentStage.mapTo(ResizeStrategy.Source))
+    }
     return stages
   }
 
+  @Nullable TargetServerGroup lookupServerGroup(Stage parentStage, String serverGroupName) {
+    def fromContext = parentStage.mapTo(ResizeStrategy.Source)
+
+    try {
+      // TODO: throw some retries in there?
+      return oortHelper.getTargetServerGroup(
+        fromContext.credentials,
+        serverGroupName,
+        fromContext.location
+      ).orElse(null)
+    } catch(Exception e) {
+      log.error('Skipping resize stage because there was an error looking up {}', serverGroupName, e)
```

Comment:
i think this error would be better in line 157/162.

that way you can differentiate between failure in asg lookup vs there legitimately not being an ASG

Reply:
chatted offline, will change the way we log when we get an empty optional